### ISSUE TITLE
Backport Power VS features to branch used by OpenShift installer

### ISSUE THREAD

**Issue by mjturek** [IBM-Cloud/terraform-provider-ibm#3574]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Power VS is getting added to the OpenShift installer as a supported platform. The OpenShift installer currently only supports terraform-plugin-sdk v1, so we cannot use recent releases. While most of the features we need have been backported to the v1.36.0 branch, there are some recent patches that have not been. Specifically, we're missing:

- [Allow passing cloud connection while creating DHCP server](https://github.com/IBM-Cloud/terraform-provider-ibm/commit/b56cefb87e60e82dc41e19d72189a54ff32cb2ac)
- [Add dns attribute to power network ](https://github.com/IBM-Cloud/terraform-provider-ibm/commit/52c7c845abdcfa889eeba68a60ee6e649c0713db)

Additionally, the branch of terraform-provider-ibm used in the installer is currently [openshift/terraform-provider-ibm@v1.26.2-openshift-2](https://github.com/openshift/terraform-provider-ibm/tree/v1.26.2-openshift-2), so a bump will be needed in the installer. IBM Cloud is a supported platform so we are not the only consumer of terraform-provider-ibm in the OpenShift installer. This means that any bump needs to be given the OK from IBM Cloud folks as well.

In short, we need to backport some changes into a branch that still used the v1 terraform-plugin-sdk (such as v1.36.0), push these updates to the openshift fork of this branch, and bump to this version in the installer.

### New or Affected Resource(s)

* ibm_pi_network
* ibm_pi_dhcp

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by yussufsh** [comment]
IMO this issue should be created in https://github.com/openshift/terraform-provider-ibm repository to back-port any changes in their fork.
I am not clear on the idea here.

**Comment by mjturek** [comment]
@yussufsh Unfortunately, https://github.com/openshift/terraform-provider-ibm does not have issues enabled.

Let me try to clarify.

In short, we need your changes backported to v1.36.0

The patches I've listed are ones we need for Power VS with openshift installer. We can only use the v1 terraform-plugin-sdk. They are not backported to a branch that uses the v1 terraform-plugin-sdk.

The v1.36.0 branch seems like the best candidate to backport to. Once that happens will need to talk with IBM Cloud IPI folks about bumping to this version in the installer.

**Comment by mjturek** [comment]
Tried backporting the first patch myself [here](https://github.com/mjturek/terraform-provider-ibm/commit/38f3b1e58e22d638e21dc9956f4c40656b2a23bf) but still need to test

**Comment by yussufsh** [comment]
I have backported the DHCP/Cloud Connection commit to https://github.com/yussufsh/terraform-provider-ibm/tree/release-1.36.0 based on the openshift forked branch.
Going to close this issue please open issue with openshift fokr or my fork  https://github.com/yussufsh/terraform-provider-ibm in future.

**Comment by mjturek** [comment]
Thanks for the help @yussufsh 

Again, can't open issues on both of those forks as issues is not enabled.
